cairo/ethereum/cancun/keth/aggregator.cairo

```diff
@@ -0,0 +1,224 @@
+struct TaskOutputHeader {
+    size: felt,  // Size including this header + actual output
+    program_hash: felt,
+}
+
+// Expected public outputs for each Keth segment.
+struct KethInitOutput {
+    // Commitment to the inputs received by `body`.
+    body_commitment_0: felt,
+    body_commitment_1: felt,
+    // Commitment to the inputs received by `teardown`.
+    teardown_commitment_0: felt,
+    teardown_commitment_1: felt,
+}
+
+struct KethBodyOutput {
+    // Commitment to the inputs received by this `body` chunk.
+    input_commitment_0: felt,
+    input_commitment_1: felt,
+    // Commitment to the outputs produced by this `body` chunk (state after its transactions).
+    post_exec_commitment_0: felt,
+    post_exec_commitment_1: felt,
+}
+
+struct KethTeardownOutput {
+    // Commitment of inputs it expected from `init`.
+    init_args_commitment_check_0: felt,
+    init_args_commitment_check_1: felt,
+    // Commitment of inputs it expected from the last `body` chunk.
+    body_args_commitment_check_0: felt,
+    body_args_commitment_check_1: felt,
+}
+
+// Core Keth STF Aggregator logic.
+// Verifies the consistency of commitments between Keth segments.
+// Outputs the segment data in the format expected by ApplicativeBootloader for comparison.
+// @param taken as hint inputs:
+//     program_input = {
+//     "keth_segment_outputs": [ [init_out], [body1_out], ..., [bodyN_out], [teardown_out] ],
+//     "keth_segment_program_hashes": { "init": H_init, "body": H_body, "teardown": H_teardown },
+//     "n_body_chunks": N
+// }
+func aggregator{output_ptr: felt*, range_check_ptr}() {
+    alloc_locals;
+
+    // Number of body chunks executed.
+    local n_body_chunks: felt;
+    // Program hashes of the Keth segments.
+    local init_program_hash: felt;
+    local body_program_hash: felt;
+    local teardown_program_hash: felt;
+
+    // Pointers to the *serialized* actual outputs of each segment.
+    local serialized_init_output: felt*;
+    local serialized_body_outputs: felt**;
+    local serialized_teardown_output: felt*;
+
+    %{aggregator_inputs%}
+
+    with_attr error_message("AssertionError: Must have at least one body chunk") {
+        assert_not_equal(n_body_chunks, 0);
+    }
+
+    let init_output: KethInitOutput* = cast(serialized_init_output, KethInitOutput*);
+    let teardown_output: KethTeardownOutput* = cast(
+        serialized_teardown_output, KethTeardownOutput*
+    );
+
+    // --- Verify Commitments ---
+    // 1. Check init output links to the first body input
+    let first_body_output: KethBodyOutput* = cast(serialized_body_outputs[0], KethBodyOutput*);
+    assert init_output.body_commitment_0 = first_body_output.input_commitment_0;
+    assert init_output.body_commitment_1 = first_body_output.input_commitment_1;
+
+    // 2. Check body chunk links recursively/iteratively
+    // Only check internal links if there are more than 1 body chunk
+    let squared_n_body_chunks = n_body_chunks * n_body_chunks;
```

Comment:
we checked tht n_body_chunks is not zero above,
here we can simply do if(n_body_chunks != 1) no?